src/core/utils/hooks/useOverflow.tsx

```diff
@@ -33,22 +35,29 @@ export const useOverflow = <T extends HTMLElement>(
 ) => {
   const containerRef = React.useRef<T>(null);
 
-  const [visibleCount, setVisibleCount] = React.useState(items.length);
-  const overflowBreakpoints = React.useRef<number[]>([]);
+  const [visibleCount, setVisibleCount] = React.useState(
+    Math.min(items.length, STARTING_ITEMS_COUNT),
+  );
 
-  React.useLayoutEffect(() => {
-    setVisibleCount(items.length);
-    overflowBreakpoints.current = [];
-  }, [items]);
+  const needsFullRerender = React.useRef(true);
 
   const [containerWidth, setContainerWidth] = React.useState<number>(0);
+  const previousContainerWidth = React.useRef<number>(0);
   const updateContainerWidth = React.useCallback(
     ({ width }) => setContainerWidth(width),
     [],
   );
   const [resizeRef, observer] = useResizeObserver<T>(updateContainerWidth);
   const resizeObserverRef = React.useRef(observer);
 
+  React.useLayoutEffect(() => {
+    // Reset when width was increased
+    if (previousContainerWidth.current < containerWidth) {
```

Comment:
This hook will run even when `disabled` is true, so maybe add `!disabled` to this if condition

Reply:
Added